client/transaction-pool/src/lib.rs

```diff
@@ -591,6 +591,11 @@ where
 		let pool = self.pool.clone();
 		let api = self.api.clone();
 
+		// do not process maintain txpool if node is out of sync
+		if tree_route.enacted().len() > 20 {
```

Comment:
Applied suggestion from https://github.com/paritytech/substrate/issues/12903#issuecomment-1363423003

I think we could still process retracted part of the `tree_route`, to notify listeners that transactions were retracted and put them back into the pool. This should not be expensive. I am referring to this part of code:
https://github.com/paritytech/substrate/blob/4e383428ffc6d4eba173ca257c1aaf53d1ec339c/client/transaction-pool/src/lib.rs#L641-L700

Reply:
As @bkchr said in the mentioned comment, tree route should not be **constructed** if the block number difference between two events is too high. Building long tree routes is expensive.

Reply:
When retracting long fork and enacting other long fork, comparing block numbers will not work (as they can be equal or within threshold). 

I'll add additional check to avoid building tree route.

How about resubmitting transactions from retracted path (and also notifying tx listeners)?  Shall it be done?

Reply:
> How about resubmitting transactions from retracted path (and also notifying tx listeners)? Shall it be done?

This is not needed. The node does not operate normally in we run into this case.

Reply:
Fixed in https://github.com/paritytech/substrate/pull/13004/commits/614960630d30cc806a41c53631f971f65c56c58e